### ISSUE TITLE
After update cannot build Android with error: Xamarin.Android.Common.targets (1392, 3): [XAGJS7009] System.InvalidOperationException: Duplicate attribute.

### ISSUE THREAD

**Issue by matteomonizza** [xamarin/xamarin-android#6463]
Build stack:

  Xamarin.Android.Common.targets(1392, 3): [XAGJS7009] System.InvalidOperationException: Duplicate attribute.
  at System.Xml.Linq.XElement.AddAttributeSkipNotify (System.Xml.Linq.XAttribute a) [0x0000e] in <7a9b77c9c73b43f99a32b1ba98596ce4>:0 
  at System.Xml.Linq.XContainer.AddContentSkipNotify (System.Object content) [0x00032] in <7a9b77c9c73b43f99a32b1ba98596ce4>:0 
  at System.Xml.Linq.XContainer.Add (System.Object content) [0x00008] in <7a9b77c9c73b43f99a32b1ba98596ce4>:0 
  at Xamarin.Android.Tasks.ManifestDocument.AddLauncherIntentElements (System.Xml.Linq.XElement activity) [0x0004e] in <256050d9b61b4848876345a052636c75>:0 
  at Xamarin.Android.Tasks.ManifestDocument+<>c__DisplayClass99_0.<ActivityFromTypeDefinition>b__1 (Android.App.ActivityAttribute aa, System.Xml.Linq.XElement element) [0x00008] in <256050d9b61b4848876345a052636c75>:0 
  at Xamarin.Android.Tasks.ManifestDocument.ToElement[TAttribute] (Mono.Cecil.TypeDefinition type, System.String name, System.Func`2[T,TResult] parser, System.Func`2[T,TResult] toElement, System.Action`2[T1,T2] update) [0x0007d] in <256050d9b61b4848876345a052636c75>:0 
  at Xamarin.Android.Tasks.ManifestDocument.ActivityFromTypeDefinition (Mono.Cecil.TypeDefinition type, System.String name, System.Int32 targetSdkVersion) [0x00043] in <256050d9b61b4848876345a052636c75>:0 
  at Xamarin.Android.Tasks.ManifestDocument.Merge (Microsoft.Build.Utilities.TaskLoggingHelper log, Java.Interop.Tools.Cecil.TypeDefinitionCache cache, System.Collections.Generic.List`1[T] subclasses, System.String applicationClass, System.Boolean embed, System.String bundledWearApplicationName, System.Collections.Generic.IEnumerable`1[T] mergedManifestDocuments) [0x004c2] in <256050d9b61b4848876345a052636c75>:0 
  at Xamarin.Android.Tasks.GenerateJavaStubs.Run (Java.Interop.Tools.Cecil.DirectoryAssemblyResolver res) [0x00766] in <256050d9b61b4848876345a052636c75>:0 
  at Xamarin.Android.Tasks.GenerateJavaStubs.RunTask () [0x0000e] in <256050d9b61b4848876345a052636c75>:0 
  at Microsoft.Android.Build.Tasks.AndroidTask.Execute () [0x00000] in <f24af698a51a478ab00657efe7dfcbde>:0 

### Version Information

=== Visual Studio Community 2019 for Mac ===

Version 8.10.12 (build 16)
Installation UUID: a69fd0e8-a0ec-4b44-b769-73b2bbcb5b6d
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000140

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.140 (2020-02/51d876a041e) (64-bit)
	Package version: 612000140

=== Roslyn (Language Service) ===

3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

=== NuGet ===

Version: 5.9.0.7134

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.402/Sdks
SDK Versions:
	5.0.402
	5.0.401
	5.0.400
	5.0.302
	5.0.301
	5.0.203
	5.0.202
	5.0.201
	5.0.103
	5.0.102
	5.0.101
	5.0.100
	3.1.414
	3.1.413
	3.1.412
	3.1.411
	3.1.410
	3.1.409
	3.1.408
	3.1.407
	3.1.406
	3.1.405
	3.1.404
	3.1.403
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.202
	3.1.200
	3.1.102
	3.1.101
	3.1.100
	3.0.101
	3.0.100
	2.1.701
	2.1.700
	2.1.505
	2.1.504
	2.1.503
	2.1.302
	2.1.301
	2.1.4
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.11
	5.0.10
	5.0.9
	5.0.8
	5.0.7
	5.0.6
	5.0.5
	5.0.4
	5.0.3
	5.0.2
	5.0.1
	5.0.0
	3.1.20
	3.1.19
	3.1.18
	3.1.17
	3.1.16
	3.1.15
	3.1.14
	3.1.13
	3.1.12
	3.1.11
	3.1.10
	3.1.9
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.2
	3.1.1
	3.1.0
	3.0.1
	3.0.0
	2.1.23
	2.1.22
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14
	2.1.13
	2.1.12
	2.1.11
	2.1.9
	2.1.8
	2.1.7
	2.1.2
	2.1.1
	2.0.5

=== .NET Core 3.1 SDK ===

SDK: 3.1.414

=== Xamarin.Profiler ===

'/Applications/Xamarin Profiler.app' not found

=== Updater ===

Version: 11

=== Xamarin Designer ===

Version: 16.11.0.39
Hash: cd672761d
Branch: remotes/origin/d16-11
Build date: 2021-10-06 20:09:18 UTC

=== Apple Developer Tools ===

Xcode 13.0 (19234)
Build 13A233

=== Xamarin.Mac ===

Version: 8.2.0.1 (Visual Studio Community)
Hash: 8fd9e6289
Branch: xcode13.1
Build date: 2021-10-26 10:13:55-0400

=== Xamarin.iOS ===

Version: 15.2.0.1 (Visual Studio Community)
Hash: 8fd9e6289
Branch: xcode13.1
Build date: 2021-10-26 10:13:56-0400

=== Xamarin.Android ===

Version: 12.0.0.3 (Visual Studio Community)
Commit: xamarin-android/d16-11/f0e3c2d
Android SDK: /Users/matteo/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		7.1 (API level 25)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.5
SDK Build Tools Version: 30.0.2

Build Information: 
Mono: c633fe9
Java.Interop: xamarin/java.interop/d16-11@476bb5b
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-11@87af37b

=== Eclipse Temurin JDK ===

Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk/Contents/Home
1.8.0_302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.10.0.13
Hash: 1b81df5
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:12 UTC

=== Android Device Manager ===

Version: 16.10.0.15
Hash: 89dcc0b
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:30 UTC

=== Build Information ===

Release ID: 810120016
Git revision: e4f3f43ac50d8e0f83a71dbcb666bb1611a661d7
Build date: 2021-10-28 12:10:44-04
Build branch: release-8.10

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.6.0 Darwin Kernel Version 20.6.0
    Wed Jun 23 00:26:31 PDT 2021
    root:xnu-7195.141.2~5/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

NuGet Package Management Extensions 0.27

**Comment by lanarchyste** [comment]
I have the same issue since the update of Visual Studio for Mac, Xamarin Android and Xamarin iOS dated 11-03-2020

```
=== Visual Studio Community 2019 for Mac ===

Version 8.10.12 (build 16)
Installation UUID: 7bd22621-390c-4f69-b605-bffab4c04cd6
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000140

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.140 (2020-02/51d876a041e) (64-bit)
	Package version: 612000140

=== Roslyn (Language Service) ===

3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

=== NuGet ===

Version: 5.9.0.7134

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.402/Sdks
SDK Versions:
	5.0.402
	5.0.401
	5.0.400
	5.0.302
	5.0.301
	5.0.203
	5.0.202
	5.0.201
	5.0.103
	3.1.414
	3.1.413
	3.1.412
	3.1.411
	3.1.410
	3.1.409
	3.1.408
	3.1.407
	3.1.406
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.11
	5.0.10
	5.0.9
	5.0.8
	5.0.7
	5.0.6
	5.0.5
	5.0.4
	5.0.3
	3.1.20
	3.1.19
	3.1.18
	3.1.17
	3.1.16
	3.1.15
	3.1.14
	3.1.13
	3.1.12

=== .NET Core 3.1 SDK ===

SDK: 3.1.414

=== Xamarin.Profiler ===

Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 13.1 (19466)
Build 13A1030d

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 15.2.0.1 (Visual Studio Community)
Hash: 8fd9e6289
Branch: xcode13.1
Build date: 2021-10-26 10:13:56-0400

=== Xamarin.Android ===

Version: 12.0.0.3 (Visual Studio Community)
Commit: xamarin-android/d16-11/f0e3c2d
Android SDK: /Users/jbinet/Library/Android/sdk
	Supported Android versions:
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 31.0.3
SDK Build Tools Version: 31.0.0

Build Information: 
Mono: c633fe9
Java.Interop: xamarin/java.interop/d16-11@476bb5b
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-11@87af37b

=== Eclipse Temurin JDK ===

Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk/Contents/Home
1.8.0_302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.10.0.13
Hash: 1b81df5
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:12 UTC

=== Android Device Manager ===

Version: 16.10.0.15
Hash: 89dcc0b
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:30 UTC

=== Xamarin Designer ===

Version: 16.11.0.39
Hash: cd672761d
Branch: remotes/origin/d16-11
Build date: 2021-10-06 20:09:18 UTC

=== Build Information ===

Release ID: 810120016
Git revision: e4f3f43ac50d8e0f83a71dbcb666bb1611a661d7
Build date: 2021-10-28 12:10:44-04
Build branch: release-8.10

=== Operating System ===

Mac OS X 10.16.0
Darwin 21.1.0 Darwin Kernel Version 21.1.0
    Wed Oct 13 17:33:23 PDT 2021
    root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

Xamarin.Forms HotReload extension 1.4.0
NuGet Package Management Extensions 0.27
Telerik Xamarin Add-in 2021.2.615
```

**Comment by matteomonizza** [comment]
I've found that the problem is the attribute Exported on the main activity. If I remove it then it builds. I've tried to leave the attribute on the other activities cause is mandatory for Android 12, and it seems it woks. Found mandatory info on [https://developer.android.com/about/versions/12/behavior-changes-12](url)

**Comment by matteomonizza** [comment]
> I have the same issue since the update of Visual Studio for Mac, Xamarin Android and Xamarin iOS dated 11-03-2020

Did you mean 11-03-2021 ?

**Comment by lanarchyste** [comment]
> > I have the same issue since the update of Visual Studio for Mac, Xamarin Android and Xamarin iOS dated 11-03-2020
> 
> Did you mean 11-03-2021 ?

Yes

**Comment by matteomonizza** [comment]
> > > I have the same issue since the update of Visual Studio for Mac, Xamarin Android and Xamarin iOS dated 11-03-2020
> > 
> > 
> > Did you mean 11-03-2021 ?
> 
> Yes

Do you use the Export attribute?

**Comment by RonnyBansemer** [comment]
I can confirm the issue setting `exported` to `true`.
```
=== Visual Studio Community 2019 for Mac ===
Version 8.10.12 (build 16)
Installation UUID: 8d4d1eac-d15f-4c87-a2c6-1203f5a71a35
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000140

=== Mono Framework MDK ===
Runtime:
	Mono 6.12.0.140 (2020-02/51d876a041e) (64-bit)
	Package version: 612000140

=== Roslyn (Language Service) ===
3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

=== NuGet ===
Version: 5.9.0.7134

=== Xamarin Designer ===
Version: 16.11.0.39
Hash: cd672761d
Branch: remotes/origin/d16-11
Build date: 2021-10-06 20:09:18 UTC

=== .NET Core SDK ===
SDK: /usr/local/share/dotnet/sdk/5.0.402/Sdks
SDK Versions:
	5.0.402
	5.0.401
	5.0.400
	5.0.302
	5.0.301
	5.0.203
	5.0.202
	5.0.201
	5.0.103
	5.0.102
	5.0.101
	5.0.100
	3.1.414
	3.1.413
	3.1.412
	3.1.411
	3.1.410
	3.1.409
	3.1.408
	3.1.407
	3.1.406
	3.1.405
	3.1.404
	3.1.403
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.202
	3.1.201
	2.1.202
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.11
	5.0.10
	5.0.9
	5.0.8
	5.0.7
	5.0.6
	5.0.5
	5.0.4
	5.0.3
	5.0.2
	5.0.1
	5.0.0
	3.1.20
	3.1.19
	3.1.18
	3.1.17
	3.1.16
	3.1.15
	3.1.14
	3.1.13
	3.1.12
	3.1.11
	3.1.10
	3.1.9
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.3
	2.1.23
	2.1.22
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17
	2.0.9

=== .NET Core 3.1 SDK ===
SDK: 3.1.414

=== Xamarin.Profiler ===
Version: 1.6.13.11
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===
Version: 11

=== Apple Developer Tools ===
Xcode 12.4 (17801)
Build 12D4e

=== Xamarin.Mac ===
Version: 8.2.0.1 (Visual Studio Community)
Hash: 8fd9e6289
Branch: xcode13.1
Build date: 2021-10-26 10:13:55-0400

=== Xamarin.iOS ===
Version: 15.2.0.1 (Visual Studio Community)
Hash: 8fd9e6289
Branch: xcode13.1
Build date: 2021-10-26 10:13:56-0400

=== Xamarin.Android ===
Version: 12.0.0.3 (Visual Studio Community)
Commit: xamarin-android/d16-11/f0e3c2d
Android SDK: /Users/ronny/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.4
SDK Build Tools Version: 30.0.2

=== Build Information:  ===
Mono: c633fe9
Java.Interop: xamarin/java.interop/d16-11@476bb5b
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-11@87af37b

=== Eclipse Temurin JDK ===
Java SDK: /Library/Java/JavaVirtualMachines/liberica-jdk-11-full.jdk/Contents/Home
2021-07-20
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===
Version: 16.10.0.13
Hash: 1b81df5
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:12 UTC

=== Android Device Manager ===
Version: 16.10.0.15
Hash: 89dcc0b
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:30 UTC

=== Build Information ===
Release ID: 810120016
Git revision: e4f3f43ac50d8e0f83a71dbcb666bb1611a661d7
Build date: 2021-10-28 12:10:44-04
Build branch: release-8.10

=== Operating System ===
Mac OS X 10.16.0
Darwin 20.3.0 Darwin Kernel Version 20.3.0
    Thu Jan 21 00:07:06 PST 2021
    root:xnu-7195.81.3~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===
DeepClean 1.2.5
.NET Core support for Mono.Debugging 8.0.5
XAML Styler 2.0.1
NuGet Package Explorer 0.6
Xamarin Extensions - Syncfusion 18.4.0.30
```

**Comment by jonathanpeppers** [comment]
This is fixed in https://github.com/xamarin/xamarin-android/commit/9f2d3bfa4fe65eeb254ebc2382e862ba0e8c43b4. It should be going out in the next Xamarin.Android release.

The workaround is to remove `Exported = true` on any `[Activity]` attributes. Hope that helps, thanks!

**Comment by aminsepahan** [comment]
for me only removing `Exported = true` from the splash screen works too

**Comment by rajeshrayudu** [comment]
I am facing the same issue, I have updated my Xamarin.Android version also, but still facing this issue.
![image](https://user-images.githubusercontent.com/15105715/140709946-272a6335-92d4-4961-9516-0a5ed587da50.png)


![image](https://user-images.githubusercontent.com/15105715/140709301-c72bd422-d98c-4415-a370-900d7b128f67.png)

In Visual Studio Updates, it is showing Eclipse Temurin JDK downloaded. I have done "Restart and Install Update" multiple times, but it is still showing update.

Can someone help me on this?

**Comment by jonathanpeppers** [comment]
@rajeshrayudu the fix is in the coming .NET 6 release, but not yet in Xamarin.Android in VS Mac.

If you need the build now, you can use one of these builds:
* https://dl.internalx.com/vsts-devdiv/Xamarin.Android/public/5414512/main/94c2a3d86a2e0e74863b57e3c5c61dbd29daa9ea/signed/Xamarin.Android.Sdk-12.1.99.75.vsix
* https://dl.internalx.com/vsts-devdiv/Xamarin.Android/public/5414512/main/94c2a3d86a2e0e74863b57e3c5c61dbd29daa9ea/xamarin.android-12.1.99.75.pkg

**Comment by lanarchyste** [comment]
Thanks for the links !

**Comment by rajeshrayudu** [comment]
Thanks @jonathanpeppers  for the links, when will it be available for Xamarin.Android in VS Mac?

**Comment by justintoth** [comment]
I can confirm that disabling Exported = true on my Splash.activity resolved the issue. I need Exported = true to support deep links, but disabling it for now gets my dev unblocked until the new Xamarin.Android is released.

**Comment by IngweLand** [comment]
This is also an issue on Azure Pipelines, windows-2019 agent

**Comment by JesperNJessen** [comment]
Removing Exported = true from my Activity yields this error:

```
Error		ADB0010: Mono.AndroidTools.InstallFailedException: Unexpected install output: Failure [INSTALL_PARSE_FAILED_MANIFEST_MALFORMED: Failed parse during installPackageLI: /data/app/vmdl1275236033.tmp/base.apk (at Binary XML file line #47): com.zensehome.zensehome.MyFirebaseMessagingService: Targeting S+ (version 31 and above) requires that an explicit value for android:exported be defined when intent filters are present]
```

Despite having the android:exported flag set on all receiver tags in my manifest

**Comment by ivanicin** [comment]
> This is also an issue on Azure Pipelines, windows-2019 agent. However, windows-2022 agent does not have it.

I came here because I had that issue that @IngweLand mentions. At least for me it appears only when targeting Android 12. I am not sure if this is a new issue or regression or something else, but I can confirm that this still affects App Center builds. So there is something that still needs to be fixed just at this point I am not sure what it is and where to submit the problem. Can someone help regarding that?

**Comment by jonathanpeppers** [comment]
So if others can confirm, the fix is in Xamarin.Android 12.1 or higher.

This version is already on the `windows-2022` Azure DevOps image, because it has Visual Studio 2022.

On older CI images, you can install a newer Xamarin.Android versions yourself. Some examples on that [here](https://github.com/jonathanpeppers/boots).

The build in VS 2022, currently is: 

* https://download.visualstudio.microsoft.com/download/pr/5dd0fffd-aabe-4e13-9f6e-6c344d31f0ac/612151fd0da7d1dd55a889a30b590e47f05e30f46cd51d0f444e785c8693972a/Xamarin.Android.Sdk-12.1.0.5.vsix

**Comment by sunkerGit** [comment]
Hi,

I'm using Mac and updated the latest Xamarin Android 12.0.0.3 on VS 2019 community. is the issue fixed?

**Comment by Kukulkano** [comment]
I have the same issue. I pulled all the recent updates and still can not compile using **Exported = true**. But I need that to let my app open specific file types. Looks like I only get _Xamarin.Android_ version _12.0.0.3_. 

I do _Check for updates..._ in Visual Studio and this is the current versions I have and get:

> === Visual Studio Community 2019 for Mac ===
> 
> Version 8.10.18 (build 11)
> Installation UUID: f8a04d18-94a1-47bd-acea-fa59a446fd89
> 	GTK+ 2.24.23 (Raleigh theme)
> 	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)
> 
> 	Package version: 612000162
> 
> === Mono Framework MDK ===
> 
> Runtime:
> 	Mono 6.12.0.162 (2020-02/2ca650f1f62) (64-bit)
> 	Package version: 612000162
> 
> === Roslyn (Language Service) ===
> 
> 3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb
> 
> === NuGet ===
> 
> Version: 5.9.0.7134
> 
> === .NET SDK (x64) ===
> 
> SDK: /usr/local/share/dotnet/sdk/5.0.405/Sdks
> SDK Versions:
> 	5.0.405
> 	5.0.401
> 	5.0.203
> 	5.0.103
> 	5.0.101
> 	3.1.416
> 	3.1.413
> 	3.1.409
> 	3.1.406
> 	3.1.402
> 	3.1.401
> 	3.1.302
> 	3.1.301
> 	3.1.300
> 	3.1.200
> MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks
> 
> === .NET Core Runtime ===
> 
> Runtime: /usr/local/share/dotnet/dotnet
> Runtime Versions:
> 	5.0.14
> 	5.0.10
> 	5.0.6
> 	5.0.3
> 	5.0.1
> 	3.1.22
> 	3.1.19
> 	3.1.15
> 	3.1.12
> 	3.1.8
> 	3.1.7
> 	3.1.6
> 	3.1.5
> 	3.1.4
> 	3.1.2
> 	2.1.23
> 	2.1.22
> 	2.1.21
> 	2.1.20
> 	2.1.19
> 	2.1.18
> 
> === .NET Core 3.1 SDK ===
> 
> SDK: 3.1.416
> 
> === .NET 5.0 SDK ===
> 
> SDK: 5.0.405
> 
> === Xamarin Designer ===
> 
> Version: 16.11.0.60
> Hash: 56f9b80b0
> Branch: remotes/origin/d16-11
> Build date: 2021-12-15 02:44:16 UTC
> 
> === Xamarin.Profiler ===
> 
> Version: 1.6.15.68
> Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler
> 
> === Updater ===
> 
> Version: 11
> 
> === Apple Developer Tools ===
> 
> Xcode 13.2.1 (19586)
> Build 13C100
> 
> === Xamarin.Mac ===
> 
> Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.
> 
> === Xamarin.Android ===
> 
> Version: 12.0.0.3 (Visual Studio Community)
> Commit: xamarin-android/d16-11/f0e3c2d
> Android SDK: /Users/buildmaster/Library/Developer/Xamarin/android-sdk-macosx
> 	Supported Android versions:
> 		None installed
> 
> SDK Tools Version: 26.1.1
> SDK Platform Tools Version: 30.0.4
> SDK Build Tools Version: 30.0.2
> 
> Build Information: 
> Mono: c633fe9
> Java.Interop: xamarin/java.interop/d16-11@476bb5b
> ProGuard: Guardsquare/proguard/v7.0.1@912d149
> SQLite: xamarin/sqlite/3.35.4@85460d3
> Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-11@87af37b
> 
> === Eclipse Temurin JDK ===
> 
> Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk/Contents/Home
> 1.8.0.302
> Android Designer EPL code available here:
> https://github.com/xamarin/AndroidDesigner.EPL
> 
> === Android SDK Manager ===
> 
> Version: 16.10.0.13
> Hash: 1b81df5
> Branch: remotes/origin/d16-10
> Build date: 2021-11-12 01:17:32 UTC
> 
> === Android Device Manager ===
> 
> Version: 16.10.0.15
> Hash: 89dcc0b
> Branch: remotes/origin/d16-10
> Build date: 2021-11-12 01:17:52 UTC
> 
> === Xamarin.iOS ===
> 
> Version: 15.4.0.0 (Visual Studio Community)
> Hash: 8fc41ae82
> Branch: xcode13.2
> Build date: 2021-12-09 01:07:56-0500
> 
> === Build Information ===
> 
> Release ID: 810180011
> Git revision: 5b68fe210d29028414f15a937514bdbd3178e774
> Build date: 2022-01-31 14:47:37-05
> Build branch: release-8.10
> 
> === Operating System ===
> 
> Mac OS X 11.4.0
> Darwin 20.5.0 Darwin Kernel Version 20.5.0
>     Sat May  8 05:10:33 PDT 2021
>     root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64
> 
> 
> 
>

**Comment by memmaker** [comment]
I am running Visual Studio 2019 for Mac and there is no update to Xamarin.Android 12.1 to be found. When will this become available?

**Comment by Kukulkano** [comment]
Thanks, that worked.

> That newer Xamarin.Android versions are only in VS Mac 2022.

Is there any good reason for not providing the updates to VS 2019?

**Comment by awesome1128** [comment]
> Exported = true

Where can I find this part?

**Comment by awesome1128** [comment]
> I've found that the problem is the attribute Exported on the main activity. If I remove it then it builds. I've tried to leave the attribute on the other activities cause is mandatory for Android 12, and it seems it woks. Found mandatory info on [https://developer.android.com/about/versions/12/behavior-changes-12](url)

Where can I find export=true part? I am new with xamarin. Please let me know. Thanks.

**Comment by justintoth** [comment]
@awesome1128 You can set it on an Activity class with an attribute.
```

[Activity (Name = "com.xxx.xxx.SplashActivity",
               Theme = "@style/Theme.RPR.NoActionBar",
               Label = "@string/application_name",
               MainLauncher = true,
               Exported = true,
               NoHistory = true)]
    public class SplashActivity : SplashActivityBase {
```

**Comment by awesome1128** [comment]
> @awesome1128 You can set it on an Activity class with an attribute.
> 
> ```
> 
> [Activity (Name = "com.xxx.xxx.SplashActivity",
>                Theme = "@style/Theme.RPR.NoActionBar",
>                Label = "@string/application_name",
>                MainLauncher = true,
>                Exported = true,
>                NoHistory = true)]
>     public class SplashActivity : SplashActivityBase {
> ```

Ah, got it. Thank you for letting me know.

**Comment by venkatsunilm** [comment]
I have VS 2022 in Windows machine. V17.1.3
I have updated Androidx and XF etc to the latest available PFB the complete details. 
Hi, I have used Exported = true in SplashActivity and in AndroidManifest in activity attribute as mentioned above. 
The problem is resolved when i delete the bin and obj folder but when i compile again i cannot install in the devices which is running above API 30 but no issues below that. It's like if I delete bin, obj and restart VS then it installs. 

Any help appreciated. :)

Microsoft Visual Studio Enterprise 2022
Version 17.1.3
VisualStudio.17.Release/17.1.3+32328.378
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

Azure App Service Tools v3.0.0   17.1.358.51495
Azure App Service Tools v3.0.0

C# Tools   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Mono Debugging for Visual Studio   17.1.10 (65f98f5)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.1.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Visual Basic Tools   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.21610.4+07b5673e4f2fa7630e78abe37f16b372353a7242
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   17.1.0.324 (d17-1@590ddb7)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.1.0.87 (remotes/origin/main@d50fb5635)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.1.8 (590388a)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   12.2.0.4 (d17-1/19b97b7)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: a5d1934
    Java.Interop: xamarin/java.interop/d17-1@7ac33610
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.37.1@0893b3b
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d17-1@db125a7


Xamarin.iOS and Xamarin.Mac SDK   15.8.0.2 (79ddd0840)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
